airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/context.py

```diff
@@ -206,6 +206,11 @@ def docker_hub_password_secret(self) -> Optional[Secret]:
             return None
         return self.dagger_client.set_secret("docker_hub_password", self.docker_hub_password)
 
+    async def connector_secrets(self):
+        if self._connector_secrets is None:
+            self._connector_secrets = await secrets.get_connector_secrets(self)
+        return self._connector_secrets
+
```

Comment:
Is this related to your changes?

Reply:
It is! We were mounting connector secrets in between steps. So I hoisted that logic into the context so when you request `connector_secrets` it mounts and gets them if it hasnt already

Reply:
Could we rename the function to `get_connector_secrets` then? Or would it make sense to make `secrets.get_connector_secrets` a method of the context class?

Reply:
Im torn on this

On one hand it belongs in `context` given how coupled it is to a `ConnectorContext`

On the other hand connector secrets is a foundational concept that deserves a separate file to hold onto the logic.

I think we gain more from it being separate.

WDYT?

Reply:
I suggest renaming it to `get_connector_secrets` as it:
* Downloads the connector secrets
* Assign the secrets to `self._connector_secrets`.

And we rename `self._connector_secrets` to `self.connector_secrets` to make it public.

I'd be confused to have a method named `connector_secrets` acting as a getter on `self._connector_secrets`

Reply:
Sounds good! Though, lets leave the internal field with the underscore. We want devs to always reference the getter and feel bad otherwise